palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/response/AuditMessage.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.palisade.response;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * Audit information for a request provided by each service.
+ * Each individual service sends a record to the Audit Service for every request that it receives.
+ * The components of the message will differ depending on which service has sent the data and if the processing was
+ * successful or not.
+ *
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AuditMessage {
+
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public final String timeStamp; //when the service processed the request
+    public final String serverIp;   //server for the service that processed the request
+    public final String serverHostname;  //server host name for the service
+    public final Context context;  //Context of the client's request
+    public final String userId;    //User Id for the client.  Can be null if there is a User
+    public final String resourceId;  //Resource Id for the client.  Can be null if there is a Resource
+    public final String errorMessage;  //Error message that occurred during the processing of the request.  Will be null if there was no issue.
```

Comment:
The idea here is that each of the AuditMessage would only hold the data that is relevant to the service that sent the data (plus optionally the error message if there is one).  This means that the AuditMessage for the palisade service will only have to know about things like userID (a String) and resourceID (a String).  It does not need to know anything about User Resource, Rules etc and any baggage associated with other services.  The AuditMessage in the audit-service is at the other extreme of this and would have to know about all of the baggage for all of the different services as it would be getting data from each of the services.

Reply:
I had added a small unit test to the AuditMessageTest class (for audit-service).  It is a test where the starting point is a Json string representing a AuditMessage from the palisade-service and the object that is being used in the deserialising is the AuditMessage for the audit-service.  This is meant to represent the data that is being sent from all of the services to the audit-service where it is centrally processed.